custom/crediblex/portfolio/loc/src/main/java/com/crediblex/fineract/portfolio/loc/api/LineOfCreditApiResource.java

```diff
@@ -0,0 +1,259 @@
+
+        return this.toApiJsonSerializer.serialize(settings, Collections.singleton(template), Collections.singleton("lineOfCredit"));
+    }
+
+
+    @GET
+    @Path("clients/{clientId}/creditlines")
+    @Consumes({ MediaType.APPLICATION_JSON })
+    @Produces({ MediaType.APPLICATION_JSON })
+    @Operation(summary = "List Line of Credits for Client", description = "Retrieves all line of credits for a specific client.")
+    @ApiResponses({
+            @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = LineOfCreditApiResourceSwagger.GetLineOfCreditsResponse.class))) })
+    public String retrieveAllForClient(@PathParam("clientId") @Parameter(description = "clientId") final Long clientId,
+            @Context final UriInfo uriInfo) {
+
+        this.context.authenticatedUser().validateHasReadPermission("LINE_OF_CREDIT");
+
+        final ApiRequestJsonSerializationSettings settings = this.apiRequestParameterHelper.process(uriInfo.getQueryParameters());
+        final Collection<LineOfCreditData> lineOfCredits = this.readPlatformService.retrieveAllLineOfCreditsForClient(clientId);
+
+        return this.toApiJsonSerializer.serialize(settings, lineOfCredits, Collections.singleton("lineOfCredit"));
+    }
+
+    @GET
+    @Path("clients/{clientId}/creditlines/{lineOfCreditId}")
+    @Consumes({ MediaType.APPLICATION_JSON })
+    @Produces({ MediaType.APPLICATION_JSON })
+    @Operation(summary = "Retrieve a Line of Credit", description = "Retrieves a specific line of credit by ID.")
+    @ApiResponses({
+            @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = LineOfCreditApiResourceSwagger.GetLineOfCreditResponse.class))) })
+    public String retrieveOne(@PathParam("lineOfCreditId") @Parameter(description = "lineOfCreditId") final Long lineOfCreditId,
+            @Context final UriInfo uriInfo) {
+
+        this.context.authenticatedUser().validateHasReadPermission("LINE_OF_CREDIT");
+
+        final ApiRequestJsonSerializationSettings settings = this.apiRequestParameterHelper.process(uriInfo.getQueryParameters());
+        final LineOfCreditData lineOfCredit = this.readPlatformService.retrieveOne(lineOfCreditId);
+
+        return this.toApiJsonSerializer.serialize(settings, lineOfCredit, Collections.singleton("lineOfCredit"));
+    }
+
+    @POST
+    @Path("creditlines")
+    @Consumes({ MediaType.APPLICATION_JSON })
+    @Produces({ MediaType.APPLICATION_JSON })
+    @Operation(summary = "Create a Line of Credit", description = "Creates a new line of credit for a client.")
+    @ApiResponses({
+            @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = LineOfCreditApiResourceSwagger.PostLineOfCreditResponse.class))) })
+    public String create(@Parameter(hidden = true) final String apiRequestBodyAsJson) {
+
```

Comment:
Make this adjustment. Use a typed object not a an apiRequestAsJson. This is being phased out even in the community.

Reply:
Refer to tax component api for examples